### PR TITLE
Document wEntitiesOptions1Table and hRoomStatus

### PR DESCRIPTION
**Pull request by kemenaran**
Based on Saver's findings, provided by @daid.

I updated @daid 's peephole-replace script, to support looking up single-digits values.

### CODE REVIEW

File: src/constants/memory/hram.asm [hunk 1]
```diff
@@ -499,11 +499,11 @@ hMapId::
...
 ; Status flags of the current room
-; 0x00 = not visited,
-; 0x04 = bos door is open,
-; 0x10 = special event occured,
-; 0x30 = don't load room entities,
-; 0x80 = visited,
+; 0x00 = not visited
+; 0x04 = boss door is open
+; 0x10 = special event occured
+; 0x30 = boss defeated
```
Comment (daid) [inline src/constants/memory/hram.asm:505]: These are slightly "wrong". 0x10, 0x20, 0x40 are just room flags, depending on the entities/events in the room how they are used.

0x80 is used to show the room as visited on the map (both the dungeon map and the overworld map)

0x01, 0x02, 0x04, 0x08 are the 4 door direction doors are open if I remember right, independend of the type of door. Not sure which bit is which direction.

File: tools/peephole-replace.py [hunk 2]
```diff
@@ -331,6 +355,26 @@ def process(self, lines):
...
     }),
+    PeepholeRule("""
+        ld   hl, wEntitiesOptions1Table
+        add  hl, de
+        set  @, [hl]
+    """, ENTITY_OPT_BITS),
+    PeepholeRule("""
+        ld   hl, wEntitiesOptions1Table
+        add  hl, bc
+        set  @, [hl]
```
Comment (daid) [inline tools/peephole-replace.py:366]: Two times the same rule?

File: tools/peephole-replace.py [hunk 3]
```diff
@@ -331,6 +355,26 @@ def process(self, lines):
...
+    """, ENTITY_OPT_BITS),
+    PeepholeRule("""
+        ld   hl, wEntitiesOptions1Table
+        add  hl, bc
```
Comment (daid) [inline tools/peephole-replace.py:370]: think you wanted `add hl, de` here, as now you have the same rule twice.

Comment (kemenaran) [reply tools/peephole-replace.py:370]: oops, indeed - thanks

**Comment by kemenaran** [comment]
I pushed a new version that supports finding a Peephole replacement using lambdas (which allows to format bitfields values).

File: src/constants/memory/hram.asm (hunk 1 above)
Comment (kemenaran) [reply src/constants/memory/hram.asm]: Thanks! I corrected the values, and added them to this PR.

File: src/constants/gameplay.asm [hunk 4]
```diff
@@ -339,6 +339,30 @@ TRADING_ITEM_NECKLACE          equ $0C
...
+; Flags can be combined. For example, visiting the first dungeon's screen (80)
+; and opening it with the key (10) would put that byte at 90.
+ROOM_STATUS_NONE            equ $00
+ROOM_STATUS_DOOR_OPEN_RIGHT equ $01
+ROOM_STATUS_DOOR_OPEN_LEFT  equ $02
+ROOM_STATUS_DOOR_OPEN_UP    equ $04
+ROOM_STATUS_DOOR_OPEN_DOWN  equ $08
+ROOM_STATUS_EVENT_1         equ $10
+ROOM_STATUS_EVENT_2         equ $20
+ROOM_STATUS_EVENT_3         equ $40
```
Comment (daid) [inline src/constants/gameplay.asm:353]: Main use of this flag seems to be the keyblocks in dungeons, but a few "scripted" events seem to use it as well on the overworld.

Comment (kemenaran) [reply src/constants/gameplay.asm:353]: One day I'll probably split better the status room flags for overworld (where they have often a precise use) from indoors (where it seems every entity and it's cousin use them for different reasons).

**Pull request merged**